pkg/admission/mutator/shoot.go

```diff
@@ -89,97 +93,57 @@ func (s *shoot) Mutate(_ context.Context, newObj, oldObj client.Object) error {
-			}
 
-			if oldShoot == nil && networkConfig.Overlay == nil {
-				networkConfig.Overlay = overlay
-			}
-
-			if oldShoot != nil && networkConfig.Overlay == nil {
-				oldNetworkConfig, err := s.decodeCalicoNetworkConfig(oldShoot.Spec.Networking.ProviderConfig)
-				if err != nil {
-					return err
-				}
-
-				if oldNetworkConfig.Overlay != nil {
-					networkConfig.Overlay = oldNetworkConfig.Overlay
-				}
-			}
+		overlayConfig := map[string]interface{}{enabledKey: false}
+		if (*shoot.Spec.Networking.Type == calicoReleaseName && EnableOverlayAsDefaultForCalico) || (*shoot.Spec.Networking.Type == ciliumReleaseName && EnableOverlayAsDefaultForCilium) {
+			overlayConfig = map[string]interface{}{enabledKey: true}
+		}
 
-			if networkConfig.Overlay != nil && !networkConfig.Overlay.Enabled && networkConfig.Overlay.CreatePodRoutes == nil {
-				networkConfig.Overlay.CreatePodRoutes = pointer.Bool(true)
-			}
+		networkConfig, err := s.decodeNetworkConfig(shoot.Spec.Networking.ProviderConfig)
+		if err != nil {
+			return err
+		}
 
-			shoot.Spec.Networking.ProviderConfig = &runtime.RawExtension{
-				Object: networkConfig,
-			}
+		if oldShoot == nil && networkConfig[overlayKey] == nil {
+			networkConfig[overlayKey] = overlayConfig
+		}
 
-		case cilium.ReleaseName:
-			overlay := &ciliumv1alpha1.Overlay{Enabled: false}
-			if EnableOverlayAsDefaultForCilium {
-				overlay = &ciliumv1alpha1.Overlay{Enabled: true}
-			}
-			networkConfig, err := s.decodeCiliumNetworkConfig(shoot.Spec.Networking.ProviderConfig)
+		if oldShoot != nil && networkConfig[overlayKey] == nil {
+			oldNetworkConfig, err := s.decodeNetworkConfig(oldShoot.Spec.Networking.ProviderConfig)
 			if err != nil {
 				return err
 			}
 
-			if oldShoot == nil && networkConfig.Overlay == nil {
-				networkConfig.Overlay = overlay
-			}
-
-			if oldShoot != nil && networkConfig.Overlay == nil {
-				oldNetworkConfig, err := s.decodeCiliumNetworkConfig(oldShoot.Spec.Networking.ProviderConfig)
-				if err != nil {
-					return err
-				}
-
-				if oldNetworkConfig.Overlay != nil {
-					networkConfig.Overlay = oldNetworkConfig.Overlay
-				}
+			if oldNetworkConfig[overlayKey] != nil {
+				networkConfig[overlayKey] = oldNetworkConfig[overlayKey]
 			}
+		}
 
-			if networkConfig.Overlay != nil && !networkConfig.Overlay.Enabled && networkConfig.Overlay.CreatePodRoutes == nil {
-				networkConfig.Overlay.CreatePodRoutes = pointer.Bool(true)
-			}
+		overlayConfig, _ = networkConfig["overlay"].(map[string]interface{})
+		if !overlayConfig["enabled"].(bool) && overlayConfig[createPodRoutesKey] == nil {
+			overlayConfig[createPodRoutesKey] = true
+			networkConfig[overlayKey] = overlayConfig
+		}
 
-			shoot.Spec.Networking.ProviderConfig = &runtime.RawExtension{
-				Object: networkConfig,
-			}
+		modifiedJSON, err := json.Marshal(networkConfig)
+		if err != nil {
+			return err
+		}
+		shoot.Spec.Networking.ProviderConfig = &runtime.RawExtension{
+			Raw: modifiedJSON,
```

Comment:
Do I remember correctly that the networking extension doesn't care at all about the `TypeMeta` information of the provided JSON ?